core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -133,22 +150,24 @@ private[storage] class BlockManagerDecommissioner(
 
     override def run(): Unit = {
       assert(conf.get(config.STORAGE_DECOMMISSION_RDD_BLOCKS_ENABLED))
-      while (!stopped && !Thread.interrupted()) {
+      while (!stopped && !stoppedRDD && !Thread.interrupted()) {
```

Comment:
Help me understand the difference b/w `stopped` and `stoppedRDD || stoppedShuffle` ? 

Can we replace `stopped` here with `stoppedShuffle` ? Why is `stoppedShuffle` not being checked in this loop ?

Reply:
This is the loop for migrating RDDs, it doesn't matter if the shuffles have encountered an error or were not enabled.

Reply:
stopped == we've stopped all migrations (e.g. Spark is shutting down)
stoppedRDD == we've stopped RDD migrations
stoppedShuffle == we've stopped Shuffle migrations

Reply:
No.

Reply:
Oh okay, stupid me: This is in the run loop of RDD migration :-P. Obviously you cannot use stoppedShuffle here.